logx/debug.go

```diff
@@ -0,0 +1,41 @@
+package logx
+
+import (
+	"flag"
+	"io/ioutil"
+	"log"
+)
+
+var (
+	// LogxDebug controls whether the debug logger is enabled or not.
+	LogxDebug = false
+)
+
+// Debug is the package logger for debug messages. Before calling Setup(), debug messages are discarded.
+var Debug = log.New(ioutil.Discard, "DEBUG: ", 0)
+
+func init() {
+	flag.BoolVar(&LogxDebug, "logx.debug", false, "Enable logx debug logging.")
+}
+
+// Setup should be called after flag.Parse(). If debug logging is enabled, Setup
+// configures the package loggers based on the LogxLevel value.
+func Setup() error {
```

Comment:
This is brittle. I would prefer that we find a solution that does not depend on remembering to call `Setup`.

Options include:
1) Having `Debug` check the status of `LogxDebug` on every output.
2) Having the flag parsing set the value of `Debug` appropriately.

I think I prefer number 2. Essentially, we are calling `Setup` here because flag parsing doesn't do enough, so let's make it do more.